packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -589,8 +589,11 @@ class FlutterErrorDetails with Diagnosticable {
     if (exception is FlutterError) {
       return exception as FlutterError;
     }
-    if (exception is AssertionError && exception.message is FlutterError) {
-      return exception.message as FlutterError;
+    if (exception is AssertionError) {
+      final AssertionError assertion = exception as AssertionError;
+      if (assertion.message is FlutterError) {
+        return assertion.message as FlutterError;
+      }
     }
     return null;
   }
```

Comment:
we could do `final Object exception = this.exception` as the top of this function to remove all the `as`es.